File: Yandex/Practicum/Algorightms_and_Data_Structures/7_sprint/D._Числа_Фибоначчи_для_взрослых/solution.py
import sys

MOD: int = 1_000_000_007

def solution() -> None:
    """
    Time Complexity O(N)
    Memory Complexity O(1)
    """
    n: int = int(sys.stdin.readline().rstrip())
    if n <= 1:
        sys.stdout.write('1\n')
        return
    a: int = 1
    b: int = 1
    for _ in range(2, n + 1):
        a, b = b, (a % MOD + b % MOD) % MOD
    sys.stdout.write(f'{b}\n')


if __name__ == '__main__':
    solution()

File: Yandex/Practicum/Algorightms_and_Data_Structures/7_sprint/D._Числа_Фибоначчи_для_взрослых/solution.cpp
#include <iostream>


void solution() {
    /*
    Time Complexity O(N)
    Memory Complexity O(1)
    */
    int n;
    std::scanf("%d", &n);
    if (n <= 1) {
        std::printf("1\n");
        return;
    }
    int MOD = 1000000007;
    int a = 1;
    int b = 1;
    for (int i = 2; i <= n; ++i) {
        int c = b;
        b = (a % MOD + b % MOD) % MOD;
        a = c;
    }
    std::printf("%d\n", b);
}


int main() {
    solution();
}